core/src/Revolution/Rest/modRestController.php

```diff
@@ -668,8 +668,8 @@ public function post()
         $properties = $this->getProperties();
 
         if (!empty($this->postRequiredFields)) {
-            if (!$this->checkRequiredFields($this->postRequiredFields)) {
-                return $this->failure($this->modx->lexicon('error'));
+            if ($result = $this->checkRequiredFields($this->postRequiredFields) !== true) {
+                return $this->failure($result);
             }
```

Comment:
I could be wrong here, but I believe your change still will not capture the validation string in case of an error. This will:
```suggestion
            $result = $this->checkRequiredFields($this->postRequiredFields);
            if ($result !== true) {
                return $this->failure($result);
            }
```

Reply:
Both code variants are valid. An assignment in the if clause is maybe not the preferred code style. This can be changed.

Reply:
The two actually return different results. In the original, `$result` will always be a boolean value while the suggested will preserve the string value if present.

Reply:
The comparison runs after the assignment according to my tests (the error message contained a string with the missing fields). See the note on: https://www.php.net/manual/en/language.operators.precedence.php

Perhaps a `assignment in condition' code quality check can be enabled. But this will disallow a lot of these code shortcuts.

Reply:
You need to test it in isolation to see what I mean. See this [sandbox test](https://onlinephp.io/c/e8595). As you can see in the second case, you can get the result you're expecting if you set off the first part of the statement in parens. Regardless, it's probably better code style wise to shy away from the one-liner.

Reply:
You are right. If I add the brackets around the assignment, another error message is added. I have to check my other code, because I was sure that this comparison is valid.

Reply:
During these checks I noticed some `if(null === $variable = $class->method())` checks in foreign code. Maybe it is valid the other way around.

Reply:
Yes, it does work the other way around, but I'd still generally advise against it. BTW, you'll need to make the same update for the put and delete checks (roughly at L748 and L823).